### PR TITLE
[Fix #219] Call execute on backend

### LINKED ISSUE #219: puma:monit:config failure
cap production puma:monit:config
rvm 1.28.0 (master) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io/]
ruby-2.3.3
ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-linux]
00:00 puma:monit:config
      Uploading /tmp/monit.conf 100.0%
Warning: `execute' should be wrapped in an `on' scope in /Users/macbookpro/.rvm/gems/ruby-2.3.1/gems/capistrano-3.8.0/lib/capistrano/dsl.rb:38.

  task :example do
    on roles(:app) do
      execute 'whoami'
    end
  end

(Backtrace restricted to imported tasks)
cap aborted!
SSHKit::Runner::ExecuteError: Exception while executing as ubuntu@replaced.us-west-2.compute.amazonaws.com: undefined method `execute' for main:Object

NoMethodError: undefined method `execute' for main:Object

Tasks: TOP => puma:monit:config
(See full trace by running task with --trace)

### PR DESCRIPTION
**Pull request by ivanovaleksey**
Hi,
I am not sure if it is a good solution but it seems to work fine for me.

What do you think @seuros?

### CODE REVIEW

File: lib/capistrano/tasks/monit.rake [hunk 1]
```diff
@@ -15,10 +15,10 @@ namespace :puma do
     task :monitor do
       on roles(fetch(:puma_role)) do
         begin
-          git_plugin.sudo_if_needed "#{fetch(:puma_monit_bin)} monitor #{puma_monit_service_name}"
+          git_plugin.sudo_if_needed "#{fetch(:puma_monit_bin)} monitor #{git_plugin.puma_monit_service_name}"
```
Comment (ivanovaleksey) [inline lib/capistrano/tasks/monit.rake]: I got an error here
```
NameError: undefined local variable or method `puma_monit_service_name' for #<SSHKit::Backend::Printer:0x007febbca2da78>
```

**Comment by seuros** [comment]
Awesome, thanks.

**Comment by ivanovaleksey** [comment]
Thanks @seuros!

I have one more question with this.
Now this task works fine but I am stucked with running monit as sudo.
How I am supposed to do it?

I add `set pry: true` in my `deploy.rb` file but when I type sudo password nothing happens.
I found [sshkit-sudo](https://github.com/kentaroi/sshkit-sudo) gem?
But should I use it? Is there another

**Comment by seuros** [comment]
we could use something like : https://github.com/seuros/capistrano-sidekiq/blob/master/lib/capistrano/tasks/monit.rake#L132

**Comment by ivanovaleksey** [comment]
Doesn't it works the same like [`sudo_if_need`](https://github.com/seuros/capistrano-puma/blob/master/lib/capistrano/puma/monit.rb#L23) in this project?

I mean that using `sudo` command initially throws an error
```
sudo stdout: Nothing written
sudo stderr: sudo: no tty present and no askpass program specified
```
Then I add `set :pty, true`, now it prompts me for sudo password.
I input the password but nothing happens. The password is fully visible which is very strange.

So the only solution I can find is using sshkit-sudo gem.
I wonder if it is fine or there is another more-capistrano-like options?
What is "the best" way to run sudo commands via Capistrano?

As far as I can see Capistrano 3 doesn't use any `sudo` command and encourages developers do not use sudo (I maybe wrong here).

**Comment by seuros** [comment]
You probably need : https://docs.oseems.com/general/operatingsystem/linux/enable-passwordless-sudo

**Comment by ivanovaleksey** [comment]
Yes, I found this as well. With passwordless sudo I stucked with file permissions (for instance `/etc/monit/monitrc`).
Maybe I did something wrong. I will try once again.

Thank you for the help!

**Comment by ivanovaleksey** [comment]
Hi @seuros, do you plan to release new gem version?

**Pull request merged**